lightning/src/ln/outbound_payment.rs

```diff
@@ -1411,16 +1431,20 @@ impl OutboundPayments {
 
 	pub(super) fn add_new_awaiting_invoice(
 		&self, payment_id: PaymentId, expiration: StaleExpiration, retry_strategy: Retry,
-		max_total_routing_fee_msat: Option<u64>
+		max_total_routing_fee_msat: Option<u64>, retryable_invoice_request: Option<RetryableInvoiceRequest>
 	) -> Result<(), ()> {
 		let mut pending_outbounds = self.pending_outbound_payments.lock().unwrap();
 		match pending_outbounds.entry(payment_id) {
 			hash_map::Entry::Occupied(_) => Err(()),
 			hash_map::Entry::Vacant(entry) => {
+				retryable_invoice_request.as_ref().map(|_|
+					self.awaiting_invoice.store(true, Ordering::Release)
+				);
```

Comment:
More idiomatic to write this as the following, rather than creating a new `Option`, modifying it, and then dropping it.

```rust
if retryable_invoice_request.is_some() {
	self.awaiting_invoice.store(true, Ordering::Release);
}
```

Reply:
I tried using `map` to make the code more cleaner, but looks like simpler is better in this case! :)
Addressed in [pr3010.26](https://github.com/shaavan/rust-lightning/commits/pr3010.26)
Thanks!